src/worker/worker.ts

```diff
@@ -0,0 +1,103 @@
+import { WorkerOptions } from "../controller/ArchipelagoController"
+import { Archipelago } from "../domain/Archipelago"
+import { IArchipelago } from "../domain/interfaces"
+import { NullLogger } from "../misc/utils"
+import { IslandUpdates, Logger, PeerPositionChange } from "../types/interfaces"
+import {
+  DisposeResponse,
+  IslandResponse,
+  IslandsCountResponse,
+  IslandsResponse,
+  WorkerMessage,
+  WorkerStatusMessage,
+} from "../types/messageTypes"
+
+const options: WorkerOptions = JSON.parse(process.argv[2])
+
+console.log(`Starting worker with parameters ${JSON.stringify(process.argv)}`)
+
+const archipelago: IArchipelago = new Archipelago(options.archipelagoParameters)
+
+const logger: Logger = options.logging ? console : NullLogger
+
+let status: "idle" | "working" = "idle"
+
+process.on("message", (message: WorkerMessage) => {
+  switch (message.type) {
+    case "apply-updates":
+      applyUpdates(message.updates)
+      break
+    case "get-islands": {
+      const response: IslandsResponse = {
+        type: "islands-response",
+        payload: archipelago.getIslands(),
+        requestId: message.requestId,
+      }
+      process.send!(response)
+      break
+    }
+    case "get-islands-count": {
+      const response: IslandsCountResponse = {
+        type: "islands-count-response",
+        payload: archipelago.getIslandsCount(),
+        requestId: message.requestId,
+      }
+      process.send!(response)
+      break
+    }
+    case "get-island": {
+      const response: IslandResponse = {
+        type: "island-response",
+        payload: archipelago.getIsland(message.islandId),
+        requestId: message.requestId,
+      }
+
+      process.send!(response)
+      break
+    }
+    case "dispose-request": {
+      const response: DisposeResponse = {
+        type: "dispose-response",
+        requestId: message.requestId,
+      }
+      process.send!(response)
+    }
+  }
+})
+
+function emitUpdates(updates: IslandUpdates) {
+  process.send!({ updates })
+}
+
+function applyUpdates({
+  positionUpdates,
+  clearUpdates,
+}: {
+  positionUpdates: PeerPositionChange[]
+  clearUpdates: string[]
+}) {
+  setStatus("working")
+  const startTime = Date.now()
+
+  logger.debug(`Processing ${positionUpdates.length} position updates and ${clearUpdates.length} clear updates`)
+  emitUpdates(archipelago.clearPeers(clearUpdates))
+  emitUpdates(archipelago.setPeersPositions(positionUpdates))
```

Comment:
This is definitely not for this PR, but we should consider handling all changes (delete & update) in one operation, and just emit one update

Reply:
Totally. I think we can merge them easily, since it is a record.